gr-blocks/lib/complex_to_interleaved_char_impl.cc

```diff
@@ -37,13 +41,11 @@ int complex_to_interleaved_char_impl::work(int noutput_items,
                                            gr_vector_const_void_star& input_items,
                                            gr_vector_void_star& output_items)
 {
-    const gr_complex* in = (const gr_complex*)input_items[0];
+    const float* in = (const float*)input_items[0];
     char* out = (char*)output_items[0];
 
-    int npairs = (d_vector ? noutput_items : noutput_items / 2);
-    for (int i = 0; i < npairs; i++) {
-        *out++ = (char)lrintf(in[i].real()); // FIXME saturate?
-        *out++ = (char)lrintf(in[i].imag());
+    for (int i = 0; i < noutput_items; i++) {
+        *out++ = (char)lrintf(in[i] * d_scalar);
```

Comment:
(1) Since we're using C++11, let's move to include `cmath` and then use `std::rint`.

(2) What happens if the abs output of `rint` is > SCHAR_MAX? Thinking we want to define this behavior, and guessing what we want is to clip. If you feel otherwise, please go with that here & elsewhere, and add whatever you decide to the documentation too.

(3) instead of a direct cast I think we want a `static_cast` here and elsewhere for the conversion.

Reply:
.. or ... just use Volk as noted above & all of this goes away ... poof!